docs/contributing/mac-deps-setup.md

```diff
@@ -0,0 +1,50 @@
+# Setting Up Development Dependencies on macOS
+
+You will need to install these tools on your machine:
+
+ - Node.js v8.11.4
+ - Python 2.7
+ - Xcode and Xcode Command Line Tools (Xcode -> Preferences -> Downloads)
+
+## Node.js
+
+There are two approaches to installing Node.js, **manual** and **managed**. If you're not sure which route to go, we recommend the managed route, but do whatever makes the most sense for your development process!
```

Comment:
>  If you're not sure which route to go, we recommend the managed route, but do whatever makes the most sense for your development process!

This sentence still doesn't feel great to me, and I'll try to elaborate on how it relates to these docs.

I'd like these guides (when we give our Windows and Linux setups the same treatment) to be more of a walkthrough than a bullet list of options, to ensure we can support as many users coming to the project as possible. While I'm still :thumbsup: on ensuring we support Node version managers, I'd prefer that we focus these guides on getting someone setup to work on Desktop quickly and clearly.

I originally had this set of instructions at the end of the setup, as a way to sanity check what you had installed.

```shellsession
$ node -v
v8.11.4

$ yarn -v
1.9.4
 
$ python --version
Python 2.7.13
```

What if we made this page more of walkthrough to address these two goals:

 - verifying you have the right tools 
 - if you don't, explain where to find what you need

This doc does go a lot of the way to achieving that, but I think we can take it further and support "users without Node" as well as "users with an existing Node install" and "users who prefer node version managers" while making it clear exactly what they need.

Here's an example of how I'd structure the Node section to flow in this way. Let me know your thoughts.

-----

## Node.js

Let's see if you have the right version of `node` installed. Open a terminal and run this command inside the Desktop source directory:

```
$ node -v
```

If you see an error about being unable to find `node`, that probably means you don't have any Node tools installed. You can install Node LTS (the version we need) from the [Node.js website](https://nodejs.org/en/download/) and restart your shell.

If you see the output `v8.11.x` (where `x` is any number), you're good to go.

If you see the output `v10.x.y` you're ahead of what we currently support. We have an outstanding issue building GitHub Desktop with Node 10, and hopefully can resolve this soon. If you don't care about the version you are running, you can install the version from the [Node.js website](https://nodejs.org/en/download/) over the top of your current install.

### I need to use different versions of Node.js in different projects!

We currently support these Node version managers: `nvm`,  `n`, `asdf-nodejs`. 

#### Configuring `nvm`

1. Install `nvm` using the instructions [here](https://github.com/creationix/nvm#install-script).

2. Within the Desktop source directory, install version of Node.js it requires:

```shellsession
$ nvm install
```

3. Ensure you are running the right version

```
$ nvm use
```

4. Verify you have the right version by running `node -v` again

```
$ node -v
```

If you see `v8.11.4`, you're good to go.

#### Configuring `n`

...

#### Configuring `asdf-nodejs`

...